src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(Storage storage, FruitTransaction transaction) {
+        storage.change(transaction.getFruit(),
+                storage.get(transaction.getFruit()) + transaction.getQuantity());
```

Comment:
Make sure to follow the encapsulation principle. The `storage` object should have methods to interact with the storage data rather than directly manipulating it from outside classes. Consider adding a method in the `Storage` class that encapsulates this logic.